### ISSUE TITLE
PENTAX / RICOH Support

### ISSUE THREAD

**Issue by skorpionet100** [drewnoakes/metadata-extractor-dotnet#319]

**Comment by skorpionet100** [comment]
Hi,
I can accept your propose to try and do that myself. Please share with me any useful information to accomplish this task.
Thanks

**Comment by drewnoakes** [comment]
@skorpionet100 sure, I'll be happy to guide you.

First, I assume this information is stored in a "makernote". These are areas in TIFF data where manufacturers can store data specific to their devices. Note that both Exif and raw files use TIFF encoding.

Some more general information about makernotes here: https://github.com/drewnoakes/metadata-extractor/wiki/Makernotes

We already have a [`PentaxMakernoteDirectory`](https://github.com/drewnoakes/metadata-extractor-dotnet/blob/master/MetadataExtractor/Formats/Exif/makernotes/PentaxMakernoteDirectory.cs) class. You may be able to extend that one. In some cases, manufacturers have multiple versions of makernotes across devices. In such cases, we create different directories for each.

Let me know if you have any questions.

**Comment by drewnoakes** [comment]
Also, are you able and willing to donate those sample images to this project? Are they your original images?

**Comment by skorpionet100** [comment]
> Also, are you able and willing to donate those sample images to this project? Are they your original images?

Sure and are my original images.

**Comment by drewnoakes** [comment]
I took a quick look at this for you. This should help:

```diff
diff --git a/MetadataExtractor/Formats/Exif/ExifTiffHandler.cs b/MetadataExtractor/Formats/Exif/ExifTiffHandler.cs
index bb52ae82..e14b368d 100644
--- a/MetadataExtractor/Formats/Exif/ExifTiffHandler.cs
+++ b/MetadataExtractor/Formats/Exif/ExifTiffHandler.cs
@@ -707,6 +707,11 @@ namespace MetadataExtractor.Formats.Exif
                     // Always in Motorola byte order
                     TiffReader.ProcessIfd(this, context.WithByteOrder(isMotorolaByteOrder: true).WithShiftedBaseOffset(makernoteOffset), 8);
                 }
+                else if (firstTenChars.Equals("PENTAX \0II", StringComparison.Ordinal))
+                {
+                    PushDirectory(new PentaxMakernoteDirectory()); // TODO use the correct directory class here
+                    TiffReader.ProcessIfd(this, context.WithByteOrder(isMotorolaByteOrder: false).WithShiftedBaseOffset(makernoteOffset), 10);
+                }
             }
             else if (string.Equals(firstTenChars, "Apple iOS\0", StringComparison.Ordinal))
             {
```

The use of `PentaxMakernoteDirectory` here is just to get things working. I think the schema of values here is new, so you'll need to create a new set of classes as described in the linked wiki page above.

Good luck!

**Comment by drewnoakes** [comment]
There may also be a `"PENTAX \0MM"` form where `isMotorolaByteOrder: true`.

This might be helpful too: https://exiv2.org/makernote.html#R10

**Comment by skorpionet100** [comment]
Hi,

can I use VS 2022? The correct way to get the code is to clone the github repository?

Thanks

**Comment by drewnoakes** [comment]
@skorpionet100 yes, sorry I missed your message. VS2022 and a git clone will get you started. Create a fork of this repo, clone that to your machine, create a new branch, commit your changes, push to your fork, create a pull request here. That's the general process. You can find nice docs online to walk you through the process if you're new to it.
